pkg/s3wrapper/client.go

```diff
@@ -161,6 +168,200 @@ func (c *S3Client) UploadFile(ctx context.Context, filePath, objectName string)
 	return c.UploadStream(ctx, reader, objectName)
 }
 
+func (c *S3Client) UploadISO(ctx context.Context, ignitionConfig, objectPrefix string) error {
+	log := logutil.FromContext(ctx, c.log)
+	objectName := fmt.Sprintf("%s.iso", objectPrefix)
+
+	// Get info from the ISO's header
+	areaOffsetBytes, areaLengthBytes, err := c.getISOHeaderInfo(log, baseObjectName)
+	if err != nil {
+		err = errors.Wrapf(err, "Failed to fetch base ISO information")
+		log.Error(err)
+		return err
+	}
+	log.Debugf("areaOffsetBytes: %d, areaLengthBytes: %d", areaOffsetBytes, areaLengthBytes)
+
+	baseObjectSize, err := c.GetObjectSizeBytes(ctx, baseObjectName)
+	if err != nil {
+		err = errors.Wrapf(err, "Failed to fetch base ISO size")
+		log.Error(err)
+		return err
+	}
+
+	multiOut, err := c.client.CreateMultipartUpload(&s3.CreateMultipartUploadInput{Bucket: aws.String(c.cfg.S3Bucket), Key: aws.String(objectName)})
+	if err != nil {
+		err = errors.Wrapf(err, "Failed to start upload for %s", objectName)
+		log.Error(err)
+		return err
+	}
+	uploadID := multiOut.UploadId
+	var completedParts []*s3.CompletedPart
+
+	// First part: copy the first part of the live ISO, until the embedded area
+	completedPart, err := c.uploadPartCopy(log, aws.Int64(1), uploadID, baseObjectName, objectName, 0, areaOffsetBytes-1)
```

Comment:
The function gets `*int64` as `partNumber`.
Idk what is `aws.Int64` type but I think the usage should be either `aws.Int64` in both places or `int64` accordingly (better IMO).

Reply:
`aws.Int64` is the same as `swag.Int64` i think it's fine to use aws infra for aws functions.